try-slang.js

```diff
@@ -228,23 +215,39 @@ async function render(timeMS)
         requestAnimationFrame(render);
 }
 
-async function printResult(outputBufferRead)
+async function printResult()
 {
-    render(0);
+    // Encode commands to do the computation
+    const encoder = device.createCommandEncoder({ label: 'compute builtin encoder' });
+    const pass = encoder.beginComputePass({ label: 'compute builtin pass' });
+
+    pass.setBindGroup(0, computePipeline.bindGroup);
+    pass.setPipeline(computePipeline.pipeline);
+    pass.dispatchWorkgroups(1, 1);
+    pass.end();
+
+    // copy output buffer back in print mode
+    encoder.copyBufferToBuffer(computePipeline.outputBuffer, 0, computePipeline.outputBufferRead, 0, computePipeline.outputBuffer.size);
+
+    // Finish encoding and submit the commands
+    const commandBuffer = encoder.finish();
+    device.queue.submit([commandBuffer]);
+
     await device.queue.onSubmittedWorkDone();
     // Read the results once the job is done
     await Promise.all([
-        outputBufferRead.mapAsync(GPUMapMode.READ),
+        computePipeline.outputBufferRead.mapAsync(GPUMapMode.READ),
```

Comment:
there is only one promise here, should be able to just `await computePipeline.outputBufferRead.mapAsync(...)`?